// angular2-image-gallery/esm2015/lib/services/image.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
export class ImageService {
    constructor() {
        this.imagesUpdatedSource = new Subject();
        this.imageSelectedIndexUpdatedSource = new Subject();
        this.showImageViewerSource = new Subject();
        this.imagesUpdated$ = this.imagesUpdatedSource.asObservable();
        this.imageSelectedIndexUpdated$ = this.imageSelectedIndexUpdatedSource.asObservable();
        this.showImageViewerChanged$ = this.showImageViewerSource.asObservable();
    }
    /**
     * @param {?} images
     * @return {?}
     */
    updateImages(images) {
        this.imagesUpdatedSource.next(images);
    }
    /**
     * @param {?} newIndex
     * @return {?}
     */
    updateSelectedImageIndex(newIndex) {
        this.imageSelectedIndexUpdatedSource.next(newIndex);
    }
    /**
     * @param {?} show
     * @return {?}
     */
    showImageViewer(show) {
        this.showImageViewerSource.next(show);
    }
}
ImageService.decorators = [
    { type: Injectable },
];
function ImageService_tsickle_Closure_declarations() {
    /** @type {?} */
    ImageService.prototype.imagesUpdatedSource;
    /** @type {?} */
    ImageService.prototype.imageSelectedIndexUpdatedSource;
    /** @type {?} */
    ImageService.prototype.showImageViewerSource;
    /** @type {?} */
    ImageService.prototype.imagesUpdated$;
    /** @type {?} */
    ImageService.prototype.imageSelectedIndexUpdated$;
    /** @type {?} */
    ImageService.prototype.showImageViewerChanged$;
}
